test/blackbox-tests/test-cases/subdir-stanza/run.t

```diff
@@ -0,0 +1,11 @@
+  $ echo "(lang dune 2.4)" > dune-project
+  $ cat >dune <<EOF
+  > (rule (with-stdout-to foo.txt (echo "bar")))
+  > (subdir
+  >  bar
+  >  (rule (with-stdout-to foo.txt (echo "bar"))))
+  > EOF
+  $ mkdir bar
+  $ touch bar/dune
```

Comment:
This is necessary because folding over all the source dirs no longer discovers all the dune files. We need a special mechanism to loop over dune files now.